File: API/Entities/BookAggregate/Book.cs
using System.ComponentModel.DataAnnotations;
using API.Entities.BaseEntities;
using API.Entities.ProductAggregate;
using Microsoft.EntityFrameworkCore;

namespace API.Entities.BookAggregate;

public class Book : BaseEntity 
{
    public int ProductId { get; set; }
    public Product Product { get; set; }
    
    public string HighlightText { get; set; }
    
    public int PublisherId { get; set; }
    public Publisher Publisher { get; set; }
    public ICollection<Author> Authors { get; set; }

    public int LanguageId { get; set; }
    public Language Language { get; set; }
    
    public int PageCount { get; set; }
    public DateTime PublicationDate { get; set; }
    public string ISBN { get; set; }
}

File: client/src/app/models/book-aggregate/book-models/book.ts
import { Publisher } from '../publisher-models/publisher';
import { Author } from '../author-models/author';
import { Language } from '../language-models/language';
import { Department } from '../../product-aggregate/department-models/department';
import { Category } from '../../product-aggregate/category-models/category';
import { Price } from '../../product-aggregate/price-list-models/price';

export interface Book {
  id: number;
  name: string;
  mainPictureUrl: string;
  authors: Author[];
  priceList: Price[];
  publicationDate: string;
  highlightText: string;
}
